EliteAPI/EliteDangerousAPI.Processing.cs

```diff
@@ -91,16 +102,21 @@ public partial class EliteDangerousAPI
         /// OnChanged indicates that an existing journal file was changed, and triggers
         /// reading the new data from the journal.
         /// </summary>
-        private void OnChanged(object source, FileSystemEventArgs e) {
-            try {
-                if (e.Name == "Status.json" ) {
+        private void OnChanged(object source, FileSystemEventArgs e)
+        {
+            try
+            {
+                if (e.Name == "Status.json")
+                {
                     ProcessStatusFile(true);
-                } else if (journalPattern.Matches(e.Name)) {
+                }
+                else if (journalPattern.Match(e.Name).Success)
+                {
                     System.IO.FileInfo newJournalFile = new System.IO.FileInfo(JournalFile.FullName);
```

Comment:
It's probably simpler here to just call ProcessJournalFile, since the name hasn't changed.
Maybe change this to:

else if (JournalFile != null && e.Name == JournalFile.Name)
{
  ProcessJournalFile()
}

And then have ProcessJournalFile call System.IO.FileInfo() and check the length.